app/code/Magento/Catalog/Model/Product/Price/Validation/TierPriceValidator.php

```diff
@@ -481,7 +481,7 @@ private function retrieveGroupValue(string $code)
             $this->customerGroupsByCode[strtolower($item->getCode())] = $item->getId();
         }
 
-        return $this->customerGroupsByCode[$code];
+        return $this->customerGroupsByCode[$code] ?? false;
```

Comment:
I believe that set of `$this->customerGroupsByCode[$code] = false;` and return of `$this->customerGroupsByCode[$code]` at the line 478 is better, because second call of `retrieveGroupValue` with same group code don't try to fetch it from database again